mlx/ops.cpp

```diff
@@ -1,4 +1,10 @@
 // Copyright © 2023-2024 Apple Inc.
+
+#ifdef _MSC_VER
+// Required for using M_PI in MSVC.
+#define _USE_MATH_DEFINES
+#endif
+
```

Comment:
I think we can keep it simple and just leave these `#defines _USE_MATH_DEFINES` ungaurded. They work w/ clang and I believe g++ as well.

Reply:
Updated.